src/octoprint/util/comm.py

```diff
@@ -3126,134 +3184,213 @@ def _enqueue_for_sending(self, command, linenumber=None, command_type=None, on_s
 			self._logger.debug("Type already in send queue: " + e.type)
 			return False
 
-	def _send_loop(self):
+	def _pool_for_freebuffer(self):
 		"""
-		The send loop is responsible of sending commands in ``self._send_queue`` over the line, if it is cleared for
-		sending (through received ``ok`` responses from the printer's firmware.
+		Waiting for free buffer in printer, and while _advanced_ok_max_line >0
+		if for some reason transaction with printer will be broken, the _advanced_ok_max_line will be setted to -1 from outside
+		return value not used
+		"""
+		self._log("pool")
+		#pool for free buffer
+		while True:
+			if( self._advanced_ok_max_line > 0
+				and self._current_line > self._advanced_ok_max_line):
+				time.sleep(0.005) #pool 5ms
+			else:
+				return True
+
+	def _wait_for_line(self):
```

Comment:
rename to `_wait_for_line_confirmation` - we aren't actually waiting for a line.